crates/circuit/src/circuit_data.rs

```diff
@@ -995,6 +996,17 @@ impl CircuitData {
         self.param_table.clear();
     }
 
+    pub fn count_ops(&mut self) -> IndexMap<String, i32> {
```

Comment:
Do you mind adding a docstring with `///` (3 slashes is a docstring in rust 2 are a comment). It's not a hard requirement since this API is only used by python, but it could be called by rust.

Reply:
Also I would change the value type to be a `usize` instead of `i32`. The `CircuitData` can contain `usize::MAX` elements and in the pathological edge case we could have an output of `{"h": usize::MAX}` assuming someone had enough local memory and constructed a circuit with 2^64 -1 hadamards.

Also I think you can avoid the string copy here because pyo3 should be able to convert a `&str` into a pystring when we return it. So it might be better to change the key type to `&str`. Although I'm not 100% certain on this, but the compiler will complain if it's not true.

```suggestion
    pub fn count_ops(&mut self) -> IndexMap<&str, usize> {
```